Mathlib/LinearAlgebra/AffineSpace/AffineSubspace/Basic.lean

```diff
@@ -0,0 +1,845 @@
+/-
+Copyright (c) 2020 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Myers
+-/
+import Mathlib.LinearAlgebra.AffineSpace.AffineEquiv
+import Mathlib.LinearAlgebra.AffineSpace.AffineSubspace.Defs
+
+/-!
+# Affine spaces
+
+This file defines affine subspaces (over modules) and the affine span of a set of points.
+
+## Main definitions
+
+* `AffineSubspace k P` is the type of affine subspaces. Unlike affine spaces, affine subspaces are
+  allowed to be empty, and lemmas that do not apply to empty affine subspaces have `Nonempty`
+  hypotheses. There is a `CompleteLattice` structure on affine subspaces.
+* `AffineSubspace.direction` gives the `Submodule` spanned by the pairwise differences of points
+  in an `AffineSubspace`. There are various lemmas relating to the set of vectors in the
+  `direction`, and relating the lattice structure on affine subspaces to that on their directions.
+* `AffineSubspace.parallel`, notation `∥`, gives the property of two affine subspaces being
+  parallel (one being a translate of the other).
+* `affineSpan` gives the affine subspace spanned by a set of points, with `vectorSpan` giving its
+  direction. The `affineSpan` is defined in terms of `spanPoints`, which gives an explicit
+  description of the points contained in the affine span; `spanPoints` itself should generally only
+  be used when that description is required, with `affineSpan` being the main definition for other
+  purposes. Two other descriptions of the affine span are proved equivalent: it is the `sInf` of
+  affine subspaces containing the points, and (if `[Nontrivial k]`) it contains exactly those points
+  that are affine combinations of points in the given set.
```

Comment:
I don't think this module doc string is accurate in this file; at least some of the referenced definitions are in `.Defs` instead.

Reply:
Fixed, I'd just duplicated the module doc and forgotten to split it.